smart-contracts/osmosis/contracts/cl-vault/tests/test-tube/admin.rs

```diff
@@ -44,3 +47,90 @@ fn admin_build_tick_cache_works() {
         .unwrap();
     assert!(verify_resp.result.is_ok());
 }
+
+#[test]
+fn admin_execute_auto_claim_works() {
+    let (app, contract_address, _, _cl_pool_id, _, admin, _) =
+        fixture_dex_router(PERFORMANCE_FEE_DEFAULT);
+    let wasm = Wasm::new(&app);
+
+    for _i in 1..10 {
+        let accounts = app
+            .init_accounts(
+                &[
+                    Coin::new(ACCOUNTS_INIT_BALANCE, DENOM_BASE),
+                    Coin::new(ACCOUNTS_INIT_BALANCE, DENOM_QUOTE),
+                    Coin::new(ACCOUNTS_INIT_BALANCE, "uosmo"),
+                ],
+                2,
+            )
+            .unwrap();
+
+        let amount_base = Uint128::new(10000);
+        let amount_quote = Uint128::new(10000);
+        let mut deposit_coins = vec![];
+        if amount_base > Uint128::zero() {
+            deposit_coins.push(Coin::new(amount_base.u128(), DENOM_BASE));
+        }
+        if amount_quote > Uint128::zero() {
+            deposit_coins.push(Coin::new(amount_quote.u128(), DENOM_QUOTE));
+        }
+        let _ = wasm
+            .execute(
+                contract_address.as_str(),
+                &ExecuteMsg::AnyDeposit {
+                    amount: amount_base,
+                    asset: DENOM_BASE.to_string(),
+                    recipient: Some(accounts[0].address()),
+                    max_slippage: Decimal::bps(MAX_SLIPPAGE_HIGH),
+                },
+                &deposit_coins,
+                &accounts[0],
+            )
+            .unwrap();
+    }
+
+    let query_resp: ActiveUsersResponse = wasm
+        .query(
+            contract_address.as_str(),
+            &QueryMsg::VaultExtension(ExtensionQueryMsg::ActiveUsers {
+                limit: 100,
+                next_token: None,
+            }),
+        )
+        .unwrap();
+
+    assert!(!query_resp.users.is_empty(), "Expected users to be present");
+
+    let users: Vec<(Addr, Uint256)> = query_resp
+        .users
+        .iter()
+        .map(|user| {
+            let addr = Addr::unchecked(user);
+            (addr, Uint256::from(100u128))
+        })
+        .collect();
+
+    let auto_claim_resp = wasm
+        .execute(
+            contract_address.as_str(),
+            &ExecuteMsg::VaultExtension(ExtensionExecuteMsg::Admin(
+                AdminExtensionExecuteMsg::AutoWithdraw { users },
+            )),
+            &[],
+            &admin,
+        )
+        .unwrap();
+
+    let has_expected_event = auto_claim_resp.events.iter().any(|event| {
```

Comment:
tests need to verify state, not events

Reply:
Changing. Yet to push a more complex test that I am gonna push soon.